components/compositing/compositor.rs

```diff
@@ -944,13 +982,13 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
     /// the root pipeline is the root content pipeline. If there is pinch zoom, the root
     /// content pipeline is wrapped in a display list that applies a pinch zoom
     /// transformation to it.
-    fn set_root_content_pipeline_handling_pinch_zoom(&self, transaction: &mut Transaction) {
+    fn set_root_content_pipeline_handling_device_scaling(&self, transaction: &mut Transaction) {
         let root_content_pipeline = match self.root_content_pipeline.id {
             Some(id) => id.to_webrender(),
             None => return,
         };
 
-        let zoom_factor = self.pinch_zoom_level();
+        let zoom_factor = self.device_pixels_per_page_pixel().0;
         if zoom_factor == 1.0 {
             transaction.set_root_pipeline(root_content_pipeline);
```

Comment:
Will this path ever be taken on HiDPI devices, given that zoom_factor now includes the HiDPI factor?